WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift

```diff
@@ -46,4 +46,30 @@ public final class SingleProductScreen: ScreenObject {
         XCTAssertTrue(app.staticTexts["TIP"].exists)
         XCTAssertTrue(app.textViews[productName].exists)
     }
+
+    public func verifyProductTypeScreenLoaded(productType: String) throws -> Self {
+        let priceLabel = NSPredicate(format: "label CONTAINS[c] %@", "price")
+        let inventoryLabel = NSPredicate(format: "label CONTAINS[c] %@", "inventory")
+        let productTypeLabel = NSPredicate(format: "label CONTAINS[c] %@", productType)
+        let addVariationLabel = NSPredicate(format: "label CONTAINS[c] %@", "add variations")
+
+        // the common fields on add product screen
+        XCTAssertTrue(app.cells["product-review-cell"].exists)
+        XCTAssertTrue(app.staticTexts.containing(productTypeLabel).firstMatch.exists)
+
+        // different product types displays different fields on add product screen
+        // this is to validate that the correct screens are displayed
+        switch productType {
+        case "physical", "virtual":
+            XCTAssertTrue(app.staticTexts.containing(priceLabel).firstMatch.exists)
+            XCTAssertTrue(app.staticTexts.containing(inventoryLabel).firstMatch.exists)
+        case "variable":
+            XCTAssertTrue(app.staticTexts.containing(addVariationLabel).firstMatch.exists)
+            XCTAssertTrue(app.staticTexts.containing(inventoryLabel).firstMatch.exists)
```

Comment:
`inventoryLabel` is present for both cases, so it can be moved to common?

```suggestion
        XCTAssertTrue(app.staticTexts.containing(inventoryLabel).firstMatch.exists)
            
        // different product types displays different fields on add product screen
        // this is to validate that the correct screens are displayed
        switch productType {
        case "physical", "virtual":
            XCTAssertTrue(app.staticTexts.containing(priceLabel).firstMatch.exists)
        case "variable":
            XCTAssertTrue(app.staticTexts.containing(addVariationLabel).firstMatch.exists)
```

Reply:
ah ok, this was written with the future in mind, the next product type (`grouped`) we introduce, `inventoryLabel` will no longer be a common field so i'll leave it as is for now.

Reply:
Thanks! This is what I thought about, just wanted to be sure 🙂